### DISCUSSION TITLE
Agent and spring boot starter together?

### DISCUSSION THREAD

**Discussion by johnvlittle** [open-telemetry/opentelemetry-java-instrumentation#12386] [Q&A]
WE have tried the new relic agent and the OT agent, both produce too much traffic (expensive) and neither instruments things like scheduled processes doing work or que listeners doing work.  We assume we need to add our own instrumentation.  We are considering micrometer (excellent documentation, mature product) or OT spring boot sdk.  

The question is, do we need both the OT anent and OT Spring boot or is it one or the other?

IF we go with only the spring boot SDK:

- will it automatically pickup the things which the agent does, such as REST calls and their dependencies (database calls and feign calls), or do we have to do that by hand? 
- Does it require a collector to use with New Relic, or can it send data directly to it?

I've just seen that the OT spring boot project only supports 2.6 and 3.1.  We are stuck on spring boot 2.0.3, so this forces us to use micrometer unfortunately.

**Comment by jeanbisutti** [comment]
You can use the OpenTelemetry Spring Boot starter without the OpenTelemetry Java agent.

> will it automatically pickup the things which the agent does, such as REST calls and their dependencies (database calls and feign calls), or do we have to do that by hand?

Yes, if it is covered by the out the box instrumentation of the OpenTelemetry Spring Boot starter: https://opentelemetry.io/docs/zero-code/java/spring-boot-starter/out-of-the-box-instrumentation/

You can also enable additional instrumentation: https://opentelemetry.io/docs/zero-code/java/spring-boot-starter/additional-instrumentations/

> Does it require a collector to use with New Relic, or can it send data directly to it?

If New Relic supports OTLP. You have to check with New Relic.

> I've just seen that the OT spring boot project only supports 2.6 and 3.1. We are stuck on spring boot 2.0.3, so this forces us to use micrometer unfortunately.

Our automatic tests are checking the features from Spring Boot 2.6. We are not sure that all the features work with Spring Boot 2.0.3. If you see an issue with Spring Boot 2.0.3, we could have a look.
